### DISCUSSION TITLE
Standard input file

### DISCUSSION THREAD

**Discussion by mark-hobbs** [alan-turing-institute/PeriPy#114] [Ideas]
The structure of a standard input file needs to be defined. It would be much better if models could be defined by a self-contained input file. This is important for reproducibility and would allow users to easily share input files. It would also allow users to create input files using any convenient method. 

Defining some of the the input parameters as functions (is_displacement_boundary etc) seems to be more complicated than necessary. It would be more suitable if these parameters were arrays defined before the model class is built. This would allow users to create an input file with boundary conditions already defined. Creating functions to define boundary conditions will become very onerous for more complex problems. It also prevents users from easily sharing input files.

**Comment by tjdodwell** [comment]
Do you have a demonstration of what this might look like. I would consider using xml / json input file.

Some people would have a config file which they parse parameters which you might want to change on the same problem (parametric studies) then and input file defining geometric inputs . . . . 

I am pretty flexible on what this looks like.
